Packs/SecneurXAnalysis/Integrations/SecneurXAnalysis/SecneurXAnalysis.py

```diff
@@ -0,0 +1,774 @@
+                if patternKey.lower() in SNX_IOC_TYPES_TO_DEMISTO_TYPES.keys():
+                    patternKey = SNX_IOC_TYPES_TO_DEMISTO_TYPES[patternKey]
+                parsed_ioc_list.append(patternKey + " : " + str(patternValue))
+
+    except Exception as e:
+        raise DemistoException(e)
+    return parsed_ioc_list
+
+
+def format_report_contents(contents):
+    try:
+        def dict_to_string(nested_dict):
+            return json.dumps(nested_dict).lstrip('{').rstrip('}').replace('\'', '').replace('\"', '')
+
+        table_contents = OrderedDict()
+        for key, val in contents.items():
+            if isinstance(val, dict):
+                table_contents[key] = dict_to_string(val)
+            elif isinstance(val, list):
+                table_values = []
+                for item in val:
+                    if isinstance(item, dict):
+                        table_values.append(dict_to_string(item))
+                    else:
+                        table_values.append(item)
+                table_contents[key] = table_values
+            else:
+                table_contents[key] = val
+        return table_contents
+    except Exception as e:
+        raise DemistoException(e)
+
+
+def parse_dbot_score(reportJson):
+    dbotJson = {}
+    try:
+        if reportJson:
+            submissionType = reportJson.get(SNXReportParser.SNX_SUBMISSION_TYPE_KEY, None)
+            verdictValue = reportJson.get(SNXReportParser.JSON_VERDICTS, None)
+            indicatorValue = None
+            verdictScore = 0
+            if submissionType == SNXResponse.SNX_FILE_KEY:
+                indicatorValue = reportJson.get(SNXReportParser.JSON_SHA256, None)
+            else:
+                indicatorValue = reportJson.get(SNXReportParser.SNX_SUBMISSION_NAME_KEY, None)
+            if verdictValue is not None and verdictValue in SNX_VERDICT_TO_DBOTSCORE.keys():
+                verdictScore = SNX_VERDICT_TO_DBOTSCORE[verdictValue]
+            dbotJson["DBotScore"] = {
+                "Indicator": indicatorValue,
+                "Type": submissionType,
+                "Score": verdictScore,
+                "Vendor": SNXResponse.SNX_PROVIDER
+            }
+
+    except Exception as e:
+        raise DemistoException(e)
+    return dbotJson
+
+
+def parse_report_entity(reportJson):
+    entityJson = {}
+    try:
+        if reportJson:
+            submissionType = reportJson.get(SNXReportParser.SNX_SUBMISSION_TYPE_KEY, None)
+            verdictValue = reportJson.get(SNXReportParser.JSON_VERDICTS, None)
+            descriptionValue = reportJson.get(SNXReportParser.JSON_TAGS, verdictValue)
+            subjectName = reportJson.get(SNXReportParser.SNX_SUBMISSION_NAME_KEY, None)
+            indicatorValue = None
+            if verdictValue == SNXResponse.SNX_MALWARE_KEY:
+                if submissionType == SNXResponse.SNX_FILE_KEY:
+                    indicatorValue = reportJson.get(SNXReportParser.JSON_SHA256, None)
```

Comment:
We need to make sure we get an `indicatorValue`, and in case we get `None` then we won't create an `indicator`